src/biokbase/installed_clients/baseclient.py

```diff
@@ -4,74 +4,63 @@
 # any changes made here will be overwritten
 #
 ############################################################
-
-from __future__ import print_function
-
-import json as _json
-import os as _os
-import random as _random
-
-import requests as _requests
-
-try:
-    from configparser import ConfigParser as _ConfigParser  # py 3
-except ImportError:
-    from ConfigParser import ConfigParser as _ConfigParser  # py 2
-
-try:
-    from urllib.parse import urlparse as _urlparse  # py3
-except ImportError:
-    from urlparse import urlparse as _urlparse  # py2
-
+import json
+import os
+import random
 import time
+import traceback
+from configparser import ConfigParser
+from typing import Any
+from urllib.parse import quote, urlparse
+
+import requests
+from requests.exceptions import ConnectionError
+from urllib3.exceptions import ProtocolError
 
 _CT = "content-type"
 _AJ = "application/json"
 _URL_SCHEME = frozenset(["http", "https"])
+_CHECK_JOB_RETRYS = 3
+SERVER_ERROR_TYPE = "Unknown"
 
 
-def _get_token(user_id, password, auth_svc):
+def _get_token(user_id: str, password: str, auth_svc: str) -> str:
+    """Given a user name and password, retrieve an auth token for that user."""
     # This is bandaid helper function until we get a full
     # KBase python auth client released
     # note that currently globus usernames, and therefore kbase usernames,
     # cannot contain non-ascii characters. In python 2, quote doesn't handle
     # unicode, so if this changes this client will need to change.
-    body = (
-        "user_id="
-        + _requests.utils.quote(user_id)
-        + "&password="
-        + _requests.utils.quote(password)
-        + "&fields=token"
-    )
-    ret = _requests.post(auth_svc, data=body, allow_redirects=True)
+    body = "user_id=" + quote(user_id) + "&password=" + quote(password) + "&fields=token"
```

Comment:
`requests.utils.quote` is just a wrapper around `url lib.parse.quote`, so use the latter as the former has been deprecated.